src/utils/get-code-from-ast.ts

```diff
@@ -95,11 +100,22 @@ export const sortImports = (
         }
     });
 
-    const { code: updatedCode } = print(ast, parserOptions);
+    let { code: updatedCode } = print(ast, parserOptions);
+    
+    if (prettierOptions.strictGrouping) {
+        // Remove extra newlines after each import statement
+        updatedCode = updatedCode.replace(
+            new RegExp(`${lineTerminator}+${importLineDenominator}${lineTerminator}+`, 'gi'),
+            lineTerminator,
+        ).replace(
+            new RegExp(importLineDenominator, 'gi'),
+            '',
+        );
```

Comment:
`lineTerminator` is the terminator character used by the parser (`\n` in most cases and by default)

`importLineDenominator` is a magic string that is added at the end of every import statement when `strictGrouping` is activated. This is at the moment the most reliable way to remove blank lines between import statements.

What these two `.replace` are doing is first replacing all the newlines surrounding the denominator to a single newline (line terminator) character - effectively removing blank lines after an import statement. It then removes any remaining `lineDenominator` magic string in case they weren't surrounded by newline characters

Reply:
Awesome! Appreciate the explanation Haw.